lib/Conversion/FIRRTLToRTL/LowerToRTL.cpp

```diff
@@ -1131,26 +1191,29 @@ LogicalResult FIRRTLLowering::visitDecl(RegOp op) {
   return success();
 }
 
-LogicalResult FIRRTLLowering::visitDecl(RegInitOp op) {
+LogicalResult FIRRTLLowering::visitDecl(RegResetOp op) {
   auto resultType = lowerType(op.result().getType());
+  Value clockVal = getLoweredValue(op.clockVal());
   Value resetSignal = getLoweredValue(op.resetSignal());
   Value resetValue = getLoweredValue(op.resetValue());
+
   if (!resultType || !resetSignal || !resetValue)
     return failure();
 
   auto regResult = builder->create<sv::RegOp>(resultType, op.nameAttr());
   setLowering(op, regResult);
 
+  inBlocksForReg(
+      clockVal, resetSignal,
+      [&]() { builder->create<sv::BPAssignOp>(regResult, resetValue); },
```

Comment:
Use a non-blocking statement here as I think that's right for an always block and it's also consistent with the rest of the logic and what the SFC does here.

```suggestion
      [&]() { builder->create<sv::PAssignOp>(regResult, resetValue); },
```

Reply:
:point_up: This is the only change that I would definitely like in there.